src/mempool.cpp

```diff
@@ -0,0 +1,96 @@
+#include "veriblock/mempool.hpp"
+
+#include "veriblock/stateless_validation.hpp"
+
+namespace altintegration {
+
+bool MemPool::submitATV(const std::vector<ATV>& atvs,
+                        const AltTree& tree,
```

Comment:
I think it is better to pass params in MemPool constructor rather than passing whole tree here